### PR TITLE
doc(message): add more message types

### PR DESCRIPTION
**Pull request by arHSM**

### CODE REVIEW

File: pages/resources/message.mdx [hunk 1]
```diff
@@ -107,16 +107,20 @@ Type `19` and `20` are only in API v8. In v6, they are still type `0`. Type 21 i
...
+| SUPPRESS_EMBEDS                         | 1 << 2  | do not include any embeds when serializing this message                           |
+| SOURCE_MESSAGE_DELETED                  | 1 << 3  | the source message for this crosspost has been deleted (via Channel Following)    |
+| URGENT                                  | 1 << 4  | this message came from the urgent message system                                  |
+| HAS_THREAD                              | 1 << 5  | this message has an associated thread, with the same id as the message            |
+| EPHEMERAL                               | 1 << 6  | this message is only visible to the user who invoked the Interaction              |
+| LOADING                                 | 1 << 7  | this message is an Interaction Response and the bot is "thinking"                 |
+| FAILED_TO_MENTION_SOME_ROLES_IN_THREAD  | 1 << 8  | Some roles were not mentioned or added to the thread.                             |
+| SHOULD_SHOW_LINK_NOT_DISCORD_WARNING    | 1 << 10 | ???                                                                               |
```
Comment (splatterxl) [inline pages/resources/message.mdx:121]:
```suggestion
| SHOULD_SHOW_LINK_NOT_DISCORD_WARNING    | 1 << 10 | Instructs the client to show an impersonation warning on any links in the message                                                                          |
```

**Review by splatterxl — CHANGES_REQUESTED** [review]
Please update the table in accordance to the contributing guidelines (not your fault just something that’s best to fix for consistency)

File: pages/resources/message.mdx [hunk 2]
```diff
@@ -107,16 +107,20 @@ Type `19` and `20` are only in API v8. In v6, they are still type `0`. Type 21 i
...
+| URGENT                                  | 1 << 4  | this message came from the urgent message system                                  |
+| HAS_THREAD                              | 1 << 5  | this message has an associated thread, with the same id as the message            |
+| EPHEMERAL                               | 1 << 6  | this message is only visible to the user who invoked the Interaction              |
+| LOADING                                 | 1 << 7  | this message is an Interaction Response and the bot is "thinking"                 |
+| FAILED_TO_MENTION_SOME_ROLES_IN_THREAD  | 1 << 8  | Some roles were not mentioned or added to the thread.                             |
+| SHOULD_SHOW_LINK_NOT_DISCORD_WARNING    | 1 << 10 | Instructs the client to show an impersonation warning on any links in the message                                                                          |
+| SUPPRESS_NOTIFICATIONS                  | 1 << 12 | this message will not trigger push and desktop notifications                      |
+| IS_VOICE_MESSAGE                        | 1 << 13 | ???                                                                               |
```
Comment (dolfies) [inline pages/resources/message.mdx:123]:
```suggestion
| IS_VOICE_MESSAGE                        | 1 << 13 | This message's audio attachments are rendered as voice messages |
```

**Review by dolfies — CHANGES_REQUESTED** [review]
Please update this file with relevant changes from [upstream](https://github.com/discord/discord-api-docs/blob/main/docs/resources/Channel.md) and reformat to the project's contribution guidelines.

File: pages/resources/message.mdx [hunk 3]
```diff
@@ -107,16 +107,20 @@ Type `19` and `20` are only in API v8. In v6, they are still type `0`. Type 21 i
...
+| SUPPRESS_EMBEDS                         | 1 << 2  | Do not include any embeds when serializing this message                           |
+| SOURCE_MESSAGE_DELETED                  | 1 << 3  | The source message for this crosspost has been deleted (via Channel Following)    |
+| URGENT                                  | 1 << 4  | This message came from the urgent message system                                  |
+| HAS_THREAD                              | 1 << 5  | This message has an associated thread, with the same id as the message            |
+| EPHEMERAL                               | 1 << 6  | This message is only visible to the user who invoked the Interaction              |
+| LOADING                                 | 1 << 7  | This message is an Interaction Response and the bot is "thinking"                 |
+| FAILED_TO_MENTION_SOME_ROLES_IN_THREAD  | 1 << 8  | Some roles were not mentioned or added to the thread.                             |
+| SHOULD_SHOW_LINK_NOT_DISCORD_WARNING    | 1 << 10 | Instructs the client to show an impersonation warning on any links in the message |
```
Comment (splatterxl) [inline pages/resources/message.mdx:121]:
```suggestion
| SHOULD_SHOW_LINK_NOT_DISCORD_WARNING    | 1 << 10 | This message contains a link that impersonates Discord |
```

File: pages/resources/message.mdx [hunk 4]
```diff
@@ -107,16 +107,20 @@ Type `19` and `20` are only in API v8. In v6, they are still type `0`. Type 21 i
...
+| SUPPRESS_EMBEDS                         | 1 << 2  | Do not include any embeds when serializing this message                           |
+| SOURCE_MESSAGE_DELETED                  | 1 << 3  | The source message for this crosspost has been deleted (via Channel Following)    |
+| URGENT                                  | 1 << 4  | This message came from the urgent message system                                  |
+| HAS_THREAD                              | 1 << 5  | This message has an associated thread, with the same id as the message            |
+| EPHEMERAL                               | 1 << 6  | This message is only visible to the user who invoked the Interaction              |
+| LOADING                                 | 1 << 7  | This message is an Interaction Response and the bot is "thinking"                 |
+| FAILED_TO_MENTION_SOME_ROLES_IN_THREAD  | 1 << 8  | Some roles were not mentioned or added to the thread.                             |
+| SHOULD_SHOW_LINK_NOT_DISCORD_WARNING    | 1 << 10 | This message contains a link that impersonates Discord                              |
```
Comment (splatterxl) [inline pages/resources/message.mdx:121]:
```suggestion
| SHOULD_SHOW_LINK_NOT_DISCORD_WARNING    | 1 << 10 | This message contains a link that impersonates Discord                             |
```

File: pages/resources/message.mdx [hunk 5]
```diff
@@ -107,16 +107,20 @@ Type `19` and `20` are only in API v8. In v6, they are still type `0`. Type 21 i
...
+| SUPPRESS_EMBEDS                         | 1 << 2  | Do not include any embeds when serializing this message                           |
+| SOURCE_MESSAGE_DELETED                  | 1 << 3  | The source message for this crosspost has been deleted (via Channel Following)    |
+| URGENT                                  | 1 << 4  | This message came from the urgent message system                                  |
+| HAS_THREAD                              | 1 << 5  | This message has an associated thread, with the same id as the message            |
+| EPHEMERAL                               | 1 << 6  | This message is only visible to the user who invoked the Interaction              |
+| LOADING                                 | 1 << 7  | This message is an Interaction Response and the bot is "thinking"                 |
+| FAILED_TO_MENTION_SOME_ROLES_IN_THREAD  | 1 << 8  | Some roles were not mentioned or added to the thread.                             |
+| SHOULD_SHOW_LINK_NOT_DISCORD_WARNING    | 1 << 10 | This message contains a link that impersonates Discord                             |
```
Comment (splatterxl) [inline pages/resources/message.mdx:121]:
```suggestion
| SHOULD_SHOW_LINK_NOT_DISCORD_WARNING    | 1 << 10 | This message contains a link that impersonates Discord                            |
```

File: pages/resources/message.mdx [hunk 6]
```diff
@@ -107,16 +107,20 @@ Type `19` and `20` are only in API v8. In v6, they are still type `0`. Type 21 i
...
+| IS_CROSSPOST                            | 1 << 1  | This message originated from a message in another channel (via Channel Following) |
+| SUPPRESS_EMBEDS                         | 1 << 2  | Do not include any embeds when serializing this message                           |
+| SOURCE_MESSAGE_DELETED                  | 1 << 3  | The source message for this crosspost has been deleted (via Channel Following)    |
+| URGENT                                  | 1 << 4  | This message came from the urgent message system                                  |
+| HAS_THREAD                              | 1 << 5  | This message has an associated thread, with the same id as the message            |
+| EPHEMERAL                               | 1 << 6  | This message is only visible to the user who invoked the Interaction              |
+| LOADING                                 | 1 << 7  | This message is an Interaction Response and the bot is "thinking"                 |
+| FAILED_TO_MENTION_SOME_ROLES_IN_THREAD  | 1 << 8  | Some roles were not mentioned or added to the thread.                             |
```
Comment (splatterxl) [inline pages/resources/message.mdx:120]:
```suggestion
| FAILED_TO_MENTION_SOME_ROLES_IN_THREAD  | 1 << 8  | Some roles were not mentioned or added to the thread                              |
```

**Review by splatterxl — APPROVED** [review]

**Pull request merged**